map/cMapEditor.cpp

```diff
@@ -100,7 +100,7 @@ void cMapEditor::createRandomField(int cell, int terrainType, int size) {
                 break;
         }
 
-        map.fixCoordinatesToBeWithinMap(x, y);
+        cPoint::split(x, y) = map.fixCoordinatesToBeWithinMap(x, y);
```

Comment:
ok let me try to unpack this:

- this makes `map.fixCoordinatesToBeWithinMap` return a `split` struct (without needing to specifiy in the function body ? ( as I see `{std::clamp(x, 1, getWidth() - 2), std::clamp(y, 1, getHeight() - 2)}`). This is a return value (not by reference)
- and `cPoint::split` basically holds 2 values by reference.

This also means a `cPoint::split{}` is something different than a `cPoint` correct? 

If so, what is the benefit of using a `split` struct over returning a `cPoint` itself?

Reply:
Ah, no, it works a bit differently.
TBH, I forgot to add an explanation for it. It's not easy C++, as it uses a temporary object with references to the target variables to put the cPoint's coordinates in. I've added an explanation. Does that explain enough?
I like to explain some details in a project like this, as we can't (and shouldn't) assume any minimum skill level of contributors, IMHO.

Reply:
Oh, and the benefit is that we can write something like:
```
cPoint::split(x, y) = getPoint()
```
 instead of:
```
auto p = getPoint();
x = p.x;
y = p.y;
```
I do think that in many cases, if we add overloads (or replace) functions that take a coordinate pair to accept cPoint, the split helper class is no longer needed.